### DISCUSSION TITLE
How to use TV-regularization

### DISCUSSION THREAD

**Discussion by bshin** [AmirMardan/PyFWI#13] [Q&A] [answered]
Hi, thanks a lot for this repo!
I'm just struggling to get TV-regularization running. How do I have to use the Regularization class and the tv function? Do you maybe have a small example? Would be grateful for some help :)

**Comment by AmirMardan** [comment, marked as answer]
Hi,

To use TV regularization for FWI, you should have a dictionary in `INPA` with the name of `'tv'` and keys as for example
```python
inpa['tv'] = {
    'az': 0.0,  # regularizatin in the z-direction
    'ax': 0.2, # regularizatin in the x-direction
    'lambda_weight': 1e-4 # regularizatin weight
    }
```

If you want to use the `Regularization` class outside of the provided FWI in PyFWI, you first create the object as 

```Python
regularization = Regularization(nx, # number of samples in the x-direction
                                nz, # number of samples in the z-direction
                                dx, # Spatial sampling rate in the x-direction
                                dz # Spatial sampling rate in the z-direction
                                 )
```
Then you should call the `cost_regularization` method of this object in the cost function as
```python
rms_reg, grad_reg = regularization.cost_regularization(m0,
                                                  tv_properties=inpa['tv'],
                                                  tikhonov_properties=None
                                                  )
```

**Reply by bshin** [reply]
Thanks a lot for the detailed explanation! Works now!
